src/rdkafka_topic.c

```diff
@@ -313,6 +313,30 @@ shptr_rd_kafka_itopic_t *rd_kafka_topic_new0 (rd_kafka_t *rk,
 	if (rkt->rkt_conf.compression_codec == RD_KAFKA_COMPRESSION_INHERIT)
 		rkt->rkt_conf.compression_codec = rk->rk_conf.compression_codec;
 
+        switch (rkt->rkt_conf.compression_codec) {
+        case RD_KAFKA_COMPRESSION_GZIP:
+                /* RD_KAFKA_COMPLEVEL_DEFAULT == Z_DEFAULT_COMPRESSION so there
```

Comment:
That's leaking abstraction, we should not rely on the value of Z_DEFAULT_COMPRESSION being equal our value, unless we explicitly it define it to be the same, which we don't (and shouldn't).

So I'd rather see us handling the default case explicitly here

Reply:
done, let me know if that's better :)